src/java/org/apache/cassandra/index/sai/disk/v1/bitpack/AbstractBlockPackedReader.java

```diff
@@ -0,0 +1,235 @@
+        {
+            // A non-exact match, which is the negative index of the first value greater than the target.
+            // For example, searching for 4 against min values [3,3,5,7] produces -2, which we convert to 2.
+            blockIndex = -blockIndex;
+        }
+
+        if (blockIndex > 0)
+        {
+            // Start at the previous block, because there could be duplicate values in the previous block.
+            // For example, with block 1: [1,2,3,3] & block 2: [3,3,5,7], binary search for 3 would find
+            // block 2, but we need to start from block 1 and search both.
+            // In case non-exact match, we need to pivot left as target is less than next block's min.
+            blockIndex--;
+        }
+
+        // Find the global (not block-specific) index of the target token, which is equivalent to its row ID:
+        lastIndex = findBlockRowID(targetValue, blockIndex, exactMatch);
+        return lastIndex >= valueCount ? -1 : lastIndex;
+    }
+
+    /**
+     *
+     * @return a positive block index for an exact match, or a negative one for a non-exact match
+     */
+    private int binarySearchBlockMinValues(long targetValue)
+    {
+        int high = Math.toIntExact(blockBitsPerValue.length) - 1;
+
+        // Assume here that we'll never move backward through the blocks:
+        int low = Math.toIntExact(lastIndex >> blockShift);
+
+        // Short-circuit the search if the target is in current block:
+        if (low + 1 <= high)
+        {
+            long cmp = Long.compare(targetValue, delta(low + 1, 0));
+
+            if (cmp == 0)
+            {
+                // We have an exact match, so return the index of the next block, which means we'll start
+                // searching from the current one and also inspect the first value of the next block.
+                return low + 1;
+            }
+            else if (cmp < 0)
+            {
+                // We're in the same block. Indicate a non-exact match, and this value will be both
+                // negated and then decremented to wind up at the current value of "low" here.
+                return -low - 1;
+            }
+
+            // The target is greater than the next block's min value, so advance to that
+            // block before starting the usual search...
+            low++;
+        }
+
+        while (low <= high)
+        {
+            int mid = low + ((high - low) >> 1);
+
+            long midVal = delta(mid, 0);
+
+            if (midVal < targetValue)
+            {
+                low = mid + 1;
+            }
+            else if (midVal > targetValue)
+            {
+                high = mid - 1;
+            }
+            else
+            {
+                // target found, but we need to check for duplicates
+                if (mid > 0 && delta(mid - 1, 0) == targetValue)
+                {
+                    // there are duplicates, pivot left
+                    high = mid - 1;
```

Comment:
nit: Is there any test that hits this path?

Reply:
As above, I have removed this code.